### PR TITLE
Update Anatomy_of_extension.md

### PR DESCRIPTION
**Pull request by IshaanAdarsh**
The PR is for reviewing the changes and suggest any for improving this part of the project

### CODE REVIEW

**Review by theory — CHANGES_REQUESTED** [review]
This PR changes a single line, but the file is over 400 lines long. Can you create PRs that include all new and modified lines you'd like us to review? Otherwise, we cannot leave inline comments on unchanged lines.

File: Quick-Start/Anatomy_of_extension.md [hunk 1]
```diff
@@ -1,418 +1,418 @@
...
+ 
+2.  **Initializing extension data**: The SQL script file can also include SQL statements to populate initial data into the extension's tables or perform any required data setup during the extension installation process. 
+ 
+3.  **Providing extension functionality**: The SQL script file contains SQL statements that implement the desired functionality of the extension. For example, it can include function definitions that perform complex calculations, data manipulation, or other operations specific to the extension's purpose. 
+ 
+4.  **Managing dependencies**: When creating a PostgreSQL extension, there might be dependencies on other extensions or objects. The SQL script file can include statements to ensure that these dependencies are properly managed, such as checking for the existence of required extensions or objects before creating or modifying them. 
+ 
+5.  **Version management**: SQL script files are often used to manage versioning and updates of the extension. By including version-specific SQL statements in the script file, developers can handle schema modifications or data migrations required when upgrading the extension to a new version.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:16]: This is right, but note that each new version of an extension should have its own upgrade script. The version updates do not live all in one script.

File: Quick-Start/Anatomy_of_extension.md [hunk 2]
```diff
@@ -1,418 +1,418 @@
...
+ 
+4.  **Managing dependencies**: When creating a PostgreSQL extension, there might be dependencies on other extensions or objects. The SQL script file can include statements to ensure that these dependencies are properly managed, such as checking for the existence of required extensions or objects before creating or modifying them. 
+ 
+5.  **Version management**: SQL script files are often used to manage versioning and updates of the extension. By including version-specific SQL statements in the script file, developers can handle schema modifications or data migrations required when upgrading the extension to a new version. 
+ 
+6.  **Reproducibility and portability**: SQL script files provide a standardized and portable way to define the extension's structure and behaviour. They can be easily shared with other developers or users, allowing them to reproduce the extension in their PostgreSQL environments.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:18]: Not sure about this. I think you'd want to distribute all the files that make up an extension, not just the SQL file(s).

File: Quick-Start/Anatomy_of_extension.md [hunk 3]
```diff
@@ -1,418 +1,418 @@
...
+ 
+5.  **Version management**: SQL script files are often used to manage versioning and updates of the extension. By including version-specific SQL statements in the script file, developers can handle schema modifications or data migrations required when upgrading the extension to a new version. 
+ 
+6.  **Reproducibility and portability**: SQL script files provide a standardized and portable way to define the extension's structure and behaviour. They can be easily shared with other developers or users, allowing them to reproduce the extension in their PostgreSQL environments. 
+ 
+7.  **Documentation and reference**: SQL script files serve as a valuable source of documentation and reference for the extension. Developers can review the script file to understand the extension's structure, dependencies, and functionality. It helps in troubleshooting, maintaining, and extending the extension in the future.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:20]: Reference yes, but documentation less so. One usually includes docs in a separate file, unless one is using embedded documentation, as with [pg_readme](https://pgxn.org/dist/pg_readme/).

File: Quick-Start/Anatomy_of_extension.md [hunk 4]
```diff
@@ -1,418 +1,418 @@
...
+ 
+6.  **Reproducibility and portability**: SQL script files provide a standardized and portable way to define the extension's structure and behaviour. They can be easily shared with other developers or users, allowing them to reproduce the extension in their PostgreSQL environments. 
+ 
+7.  **Documentation and reference**: SQL script files serve as a valuable source of documentation and reference for the extension. Developers can review the script file to understand the extension's structure, dependencies, and functionality. It helps in troubleshooting, maintaining, and extending the extension in the future. 
+ 
+In summary, a SQL script file in PostgreSQL extensions plays a vital role in defining the extension's objects, behaviour, and functionality. It ensures the proper installation, initialization, and operation of the extension within the PostgreSQL database. It also serves as a documentation and reference source for developers working with the extension.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:22]: No need to repeat yourself here.

File: Quick-Start/Anatomy_of_extension.md [hunk 5]
```diff
@@ -1,418 +1,418 @@
...
+ 
+6.  **Add necessary comments:** Include comments within the script file to provide explanations or clarify the purpose of specific SQL statements. This helps other developers understand the code and makes it easier to maintain and troubleshoot the extension. 
+ 
+7.  **Test the SQL script:** Before finalizing the SQL script file, it's essential to test it to ensure that the SQL statements execute successfully without any errors. Use a tool like the `psql` command-line client or a PostgreSQL GUI to execute the script and verify its functionality. 
+ 
+By following this quick start guide, you can create a well-structured and functional SQL script file for your PostgreSQL extension. It helps developers define the extension's behaviour, create necessary objects, and implement the desired functionality within the PostgreSQL database. 
+ 
+# 2) Control File: 
+ 
+A Control file, also known as a control file template or control file script, is an essential component of a PostgreSQL extension. It provides metadata and configuration information necessary for the installation, upgrade, and management of the extension.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:45]: I don't think you need to list the other names the control file is known as. I've not heard of any of these myself.

File: Quick-Start/Anatomy_of_extension.md [hunk 6]
```diff
@@ -1,418 +1,418 @@
...
+ 
+The role of a Control file in PostgreSQL extensions is as follows: 
+ 
+1.  **Extension metadata**: The Control file contains metadata about the extension, such as its name, version, description, author, and license. This information helps users and administrators understand the purpose and characteristics of the extension. 
+ 
+2.  **Dependency management**: The Control file allows you to specify any dependencies that the extension has on other extensions or PostgreSQL versions. This ensures that the required dependencies are present before the extension can be installed or upgraded. 
+ 
+3.  **Installation instructions**: The Control file provides instructions on how to install the extension. It specifies the SQL script file(s) that need to be executed during installation, as well as any additional installation steps or considerations. 
+ 
+4.  **Upgrade instructions**: If the extension supports upgrading from a previous version, the Control file includes instructions on how to perform the upgrade process. This may involve executing SQL scripts to migrate data, modify existing objects, or handle any other necessary changes.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:55]: I don't think this is quite right. The control file can contain information to help Postgres update an extension to a new version, but doesn't contain instructions for a person to follow.

File: Quick-Start/Anatomy_of_extension.md [hunk 7]
```diff
@@ -1,418 +1,418 @@
...
+ 
+2.  **Dependency management**: The Control file allows you to specify any dependencies that the extension has on other extensions or PostgreSQL versions. This ensures that the required dependencies are present before the extension can be installed or upgraded. 
+ 
+3.  **Installation instructions**: The Control file provides instructions on how to install the extension. It specifies the SQL script file(s) that need to be executed during installation, as well as any additional installation steps or considerations. 
+ 
+4.  **Upgrade instructions**: If the extension supports upgrading from a previous version, the Control file includes instructions on how to perform the upgrade process. This may involve executing SQL scripts to migrate data, modify existing objects, or handle any other necessary changes. 
+ 
+5.  **Uninstallation instructions**: The Control file defines the steps for uninstalling the extension. It specifies the SQL script file(s) or commands needed to remove the extension's objects and revert any changes made during installation.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:57]: This is not necessary. An extension is a packaged thing, so `DROP EXTENSION` drops everything that needs dropping.

File: Quick-Start/Anatomy_of_extension.md [hunk 8]
```diff
@@ -1,418 +1,418 @@
...
+ 
+3.  **Installation instructions**: The Control file provides instructions on how to install the extension. It specifies the SQL script file(s) that need to be executed during installation, as well as any additional installation steps or considerations. 
+ 
+4.  **Upgrade instructions**: If the extension supports upgrading from a previous version, the Control file includes instructions on how to perform the upgrade process. This may involve executing SQL scripts to migrate data, modify existing objects, or handle any other necessary changes. 
+ 
+5.  **Uninstallation instructions**: The Control file defines the steps for uninstalling the extension. It specifies the SQL script file(s) or commands needed to remove the extension's objects and revert any changes made during installation. 
+ 
+6.  **Configuration options**: The Control file allows you to define custom configuration options for the extension. These options can be set by users or administrators to modify the behaviour or settings of the extension according to their needs.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:59]: No configuration options are in the control file that I can recall. Are you referring to [configuration tables](https://www.postgresql.org/docs/15/extend-extensions.html#EXTEND-EXTENSIONS-CONFIG-TABLES)? They're managed by the SQL script that creates an extension.

File: Quick-Start/Anatomy_of_extension.md [hunk 9]
```diff
@@ -1,418 +1,418 @@
...
+ 
+5.  **Uninstallation instructions**: The Control file defines the steps for uninstalling the extension. It specifies the SQL script file(s) or commands needed to remove the extension's objects and revert any changes made during installation. 
+ 
+6.  **Configuration options**: The Control file allows you to define custom configuration options for the extension. These options can be set by users or administrators to modify the behaviour or settings of the extension according to their needs. 
+ 
+7.  **Permissions and privileges**: The Control file can specify the required permissions and privileges that users must have to install, upgrade, or use the extension effectively. This helps ensure proper security and access control for the extension's functionality.  
+ 
+8.  **Documentation and reference**: The Control file often includes a section for additional documentation or references related to the extension. This can include links to external resources, tutorials, examples, or any other relevant information that helps users understand and utilize the extension effectively.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:63]: Are you referring to the `comment` field? I don't see anywhere else where docs might go. Again, they generally go into a separate README file.

File: Quick-Start/Anatomy_of_extension.md [hunk 10]
```diff
@@ -1,418 +1,418 @@
...
+ 
+6.  **Configuration options**: The Control file allows you to define custom configuration options for the extension. These options can be set by users or administrators to modify the behaviour or settings of the extension according to their needs. 
+ 
+7.  **Permissions and privileges**: The Control file can specify the required permissions and privileges that users must have to install, upgrade, or use the extension effectively. This helps ensure proper security and access control for the extension's functionality.  
+ 
+8.  **Documentation and reference**: The Control file often includes a section for additional documentation or references related to the extension. This can include links to external resources, tutorials, examples, or any other relevant information that helps users understand and utilize the extension effectively. 
+ 
+In summary, a Control file in PostgreSQL extensions serves as a crucial component that provides metadata, installation instructions, upgrade guidelines, configuration options, and other relevant information about the extension. It facilitates the installation, management, and usage of the extension in a PostgreSQL environment.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:65]: Why the repetition here?

File: Quick-Start/Anatomy_of_extension.md [hunk 11]
```diff
@@ -1,418 +1,418 @@
...
+ 
+8.  **Documentation and reference**: The Control file often includes a section for additional documentation or references related to the extension. This can include links to external resources, tutorials, examples, or any other relevant information that helps users understand and utilize the extension effectively. 
+ 
+In summary, a Control file in PostgreSQL extensions serves as a crucial component that provides metadata, installation instructions, upgrade guidelines, configuration options, and other relevant information about the extension. It facilitates the installation, management, and usage of the extension in a PostgreSQL environment. 
+ 
+## Creating PostgreSQL extension: 
+When creating a PostgreSQL extension, writing a Control file is an essential step. The Control file provides crucial metadata and instructions for the installation, upgrade, and management of the extension. Here's a quick start guide on how to write a Control file for your PostgreSQL extension: 
+ 
+1.  **Choose a filename**: The Control file is typically named "control" or "your_extension_name.control". Make sure to use the ".control" file extension.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:70]: I believe it needs to be named `$extension_name.control`. Give [the docs](https://www.postgresql.org/docs/15/extend-extensions.html) a read for the precise requirements.

File: Quick-Start/Anatomy_of_extension.md [hunk 12]
```diff
@@ -1,418 +1,418 @@
...
+ 
+In summary, a Control file in PostgreSQL extensions serves as a crucial component that provides metadata, installation instructions, upgrade guidelines, configuration options, and other relevant information about the extension. It facilitates the installation, management, and usage of the extension in a PostgreSQL environment. 
+ 
+## Creating PostgreSQL extension: 
+When creating a PostgreSQL extension, writing a Control file is an essential step. The Control file provides crucial metadata and instructions for the installation, upgrade, and management of the extension. Here's a quick start guide on how to write a Control file for your PostgreSQL extension: 
+ 
+1.  **Choose a filename**: The Control file is typically named "control" or "your_extension_name.control". Make sure to use the ".control" file extension. 
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information:
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:72]: You should link to the full documentation for the control file fields.

File: Quick-Start/Anatomy_of_extension.md [hunk 13]
```diff
@@ -1,418 +1,418 @@
...
+## Creating PostgreSQL extension: 
+When creating a PostgreSQL extension, writing a Control file is an essential step. The Control file provides crucial metadata and instructions for the installation, upgrade, and management of the extension. Here's a quick start guide on how to write a Control file for your PostgreSQL extension: 
+ 
+1.  **Choose a filename**: The Control file is typically named "control" or "your_extension_name.control". Make sure to use the ".control" file extension. 
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information: 
+ 
+    -   **comment**: A brief description of the extension. 
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:76]: If there is one.

File: Quick-Start/Anatomy_of_extension.md [hunk 14]
```diff
@@ -1,418 +1,418 @@
...
+When creating a PostgreSQL extension, writing a Control file is an essential step. The Control file provides crucial metadata and instructions for the installation, upgrade, and management of the extension. Here's a quick start guide on how to write a Control file for your PostgreSQL extension: 
+ 
+1.  **Choose a filename**: The Control file is typically named "control" or "your_extension_name.control". Make sure to use the ".control" file extension. 
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information: 
+ 
+    -   **comment**: A brief description of the extension. 
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file. 
+    -   **relocatable**: Specify whether the extension can be relocated to a different directory.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:77]: Not a different directory, but a different _schema_. Please consult [the docs](https://www.postgresql.org/docs/current/extend-extensions.html) for the most up-to-date information.

File: Quick-Start/Anatomy_of_extension.md [hunk 15]
```diff
@@ -1,418 +1,418 @@
...
+ 
+1.  **Choose a filename**: The Control file is typically named "control" or "your_extension_name.control". Make sure to use the ".control" file extension. 
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information: 
+ 
+    -   **comment**: A brief description of the extension. 
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file. 
+    -   **relocatable**: Specify whether the extension can be relocated to a different directory. 
+    -   **schema**: The name of the schema where the extension objects will be created.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:78]: If a specific schema name is required.

File: Quick-Start/Anatomy_of_extension.md [hunk 16]
```diff
@@ -1,418 +1,418 @@
...
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information: 
+ 
+    -   **comment**: A brief description of the extension. 
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file. 
+    -   **relocatable**: Specify whether the extension can be relocated to a different directory. 
+    -   **schema**: The name of the schema where the extension objects will be created. 
+    -   **requires**: List any dependencies your extension has on other extensions or specific PostgreSQL versions. 
+3.  **Add extension options**: If your extension has custom configuration options, you can define them using the `--->` syntax. Each option should include a name, a default value, a description, and any other necessary attributes.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:80]: I have never heard of this. Can you link to the docs?

File: Quick-Start/Anatomy_of_extension.md [hunk 17]
```diff
@@ -1,418 +1,418 @@
...
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file. 
+    -   **relocatable**: Specify whether the extension can be relocated to a different directory. 
+    -   **schema**: The name of the schema where the extension objects will be created. 
+    -   **requires**: List any dependencies your extension has on other extensions or specific PostgreSQL versions. 
+3.  **Add extension options**: If your extension has custom configuration options, you can define them using the `--->` syntax. Each option should include a name, a default value, a description, and any other necessary attributes. 
+ 
+4.  **Specify extension scripts**: In the Control file, you need to specify which scripts are associated with your extension. These scripts include the SQL script file(s) responsible for creating or modifying the extension's objects. Use the `script` keyword followed by the script type (`pre-install`, `install`, `post-install`, `pre-upgrade`, `upgrade`, `post-upgrade`, `pre-uninstall`, `uninstall`, `post-uninstall`) and the path to the script file.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:82]: I haven't heard of this, either. Can you link to your source, please?

File: Quick-Start/Anatomy_of_extension.md [hunk 18]
```diff
@@ -1,418 +1,418 @@
...
+    -   **schema**: The name of the schema where the extension objects will be created. 
+    -   **requires**: List any dependencies your extension has on other extensions or specific PostgreSQL versions. 
+3.  **Add extension options**: If your extension has custom configuration options, you can define them using the `--->` syntax. Each option should include a name, a default value, a description, and any other necessary attributes. 
+ 
+4.  **Specify extension scripts**: In the Control file, you need to specify which scripts are associated with your extension. These scripts include the SQL script file(s) responsible for creating or modifying the extension's objects. Use the `script` keyword followed by the script type (`pre-install`, `install`, `post-install`, `pre-upgrade`, `upgrade`, `post-upgrade`, `pre-uninstall`, `uninstall`, `post-uninstall`) and the path to the script file. 
+ 
+5.  **Include documentation**: Provide relevant documentation or references related to your extension. This can include links to external resources, examples, or any additional information that helps users understand and use your extension effectively.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:84]: I don't see this in the docs.

File: Quick-Start/Anatomy_of_extension.md [hunk 19]
```diff
@@ -1,418 +1,418 @@
...
+-   Enabling users and administrators to easily understand the purpose and requirements of the extension. 
+-   Facilitating the installation, upgrade, and management of the extension by specifying dependencies and associated scripts. 
+-   Allowing customization through extension options. 
+-   Providing a reference point for documentation and additional resources. 
+
+Remember to follow the specific format and guidelines defined by PostgreSQL when writing the Control file. The Control file plays a crucial role in the successful deployment and usage of your PostgreSQL extension. 
+
+
+# 3) Code File: 
+Code File: A code file in the context of PostgreSQL extensions refers to the source code file that contains the implementation of the extension's functionality. It is typically written in the C programming language and is responsible for defining the extension's functions, data types, operators, and other database entities.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:100]: This is only true for extensions that depend on an external compiler. SQL, PL/pgSQL, PL/Perl, etc. extensions do not require a source file separate from the SQL file.

File: Quick-Start/Anatomy_of_extension.md [hunk 20]
```diff
@@ -1,418 +1,418 @@
...
+} 
+ 
+``` 
+ 
+In this example: 
+-   The code file includes necessary header files (`postgres.h` and `fmgr.h`) that provide access to the PostgreSQL Extension API. 
+-   `PG_MODULE_MAGIC` is a macro that indicates the compatibility version of the PostgreSQL Extension API. It must be included in the code file. 
+-   `PG_FUNCTION_INFO_V1` is a macro that defines the information for an extension function. It is used to declare the function to the PostgreSQL server. 
+-   `my_extension_func` is the implementation of the extension's function. It takes two integer arguments, adds them together, and returns the result. 
+-   `PG_GETARG_INT32` is a macro that retrieves the function arguments of type `int32` (integer).
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:138]: By argument position (zero-indexed).

File: Quick-Start/Anatomy_of_extension.md [hunk 21]
```diff
@@ -1,418 +1,418 @@
...
+ 
+``` 
+ 
+In this example: 
+-   The code file includes necessary header files (`postgres.h` and `fmgr.h`) that provide access to the PostgreSQL Extension API. 
+-   `PG_MODULE_MAGIC` is a macro that indicates the compatibility version of the PostgreSQL Extension API. It must be included in the code file. 
+-   `PG_FUNCTION_INFO_V1` is a macro that defines the information for an extension function. It is used to declare the function to the PostgreSQL server. 
+-   `my_extension_func` is the implementation of the extension's function. It takes two integer arguments, adds them together, and returns the result. 
+-   `PG_GETARG_INT32` is a macro that retrieves the function arguments of type `int32` (integer).   
+-   `PG_RETURN_INT32` is a macro that sets the return value of the function to an `int32` value.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:139]: You should link to the documentation for these and other macros.

File: Quick-Start/Anatomy_of_extension.md [hunk 22]
```diff
@@ -1,418 +1,418 @@
...
+-   `my_extension_func` is the implementation of the extension's function. It takes two integer arguments, adds them together, and returns the result. 
+-   `PG_GETARG_INT32` is a macro that retrieves the function arguments of type `int32` (integer).   
+-   `PG_RETURN_INT32` is a macro that sets the return value of the function to an `int32` value. 
+    
+The code file can include additional functions, data types, operators, and other database entities as required by the extension's functionality. It provides the foundation for extending PostgreSQL and allows developers to define custom behaviour that can be executed within the database. 
+ 
+## Creating Code Files for PostgreSQL Extensions: 
+Code Files: The code files contain the actual implementation of the extension's functionality. These files are typically written in a programming language such as C, C++, Python, or any other supported language. The code files interact with the database through the SQL interface and provide additional features and capabilities. Create a quick start on how to go about writing Code Files (how it will help the developer)while creating an Extension in Postgres.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:144]: Non-SQL source code does not use the SQL interface, but the Postgres APIs such as the macro examples above. It gets compiled into a shared object library that you can then reference from the SQL script.

File: Quick-Start/Anatomy_of_extension.md [hunk 23]
```diff
@@ -1,418 +1,418 @@
...
+ 
+## Creating Code Files for PostgreSQL Extensions: 
+Code Files: The code files contain the actual implementation of the extension's functionality. These files are typically written in a programming language such as C, C++, Python, or any other supported language. The code files interact with the database through the SQL interface and provide additional features and capabilities. Create a quick start on how to go about writing Code Files (how it will help the developer)while creating an Extension in Postgres. 
+
+Sure! Here's a quick start guide on writing code files for a PostgreSQL extension: 
+
+1.  **Choose a Programming Language:** Decide on the programming language you want to use for implementing your PostgreSQL extension. Common choices include C, C++, and Python. Ensure that your chosen language is supported by PostgreSQL and has the necessary libraries and bindings for interacting with the database.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:148]: This bit right her obfuscates a _ton_ of information one needs to use an of these languages. If you plan to cover those details later, I suggest you link to them here.

File: Quick-Start/Anatomy_of_extension.md [hunk 24]
```diff
@@ -1,418 +1,418 @@
...
+
+1.  **Choose a Programming Language:** Decide on the programming language you want to use for implementing your PostgreSQL extension. Common choices include C, C++, and Python. Ensure that your chosen language is supported by PostgreSQL and has the necessary libraries and bindings for interacting with the database.   
+
+2.  **Set Up the Development Environment:** Set up your development environment with the required tools and dependencies. This typically involves installing the necessary compilers, development libraries, and PostgreSQL development packages. 
+
+3.  **Create the Code File:** Create a new code file with a descriptive name that reflects the purpose of your extension. For example, if your extension is called "my\_extension," you could create a file named "my\_extension.c" for a C-based extension or "my\_extension.py" for a Python-based extension. 
+
+4.  **Include Required Header Files:** In the code file, include the necessary header files that provide access to the PostgreSQL Extension API and any other libraries or modules you will be using. The specific header files may vary depending on the programming language and libraries you are using.
```
Comment (theory) [inline Quick-Start/Anatomy_of_extension.md:154]: For non-C languages it's usually some sort of package import. But even that's not necessary for many languages. [PL/Perl](https://www.postgresql.org/docs/current/plperl.html), for example, just has various functions available to it. In fact, All of the `PL/*` languages don't go into a separate source file, but into the SQL script.

**Review by theory — CHANGES_REQUESTED** [review]
I got about halfway through this document — it's long! — and left some comments and corrections. So let's start with those.

However I think you might be better served by putting all this documentation on hold and actually writing and packaging a dead simple SQL-only extension.

File: Quick-Start/Anatomy_of_extension.md (hunk 23 above)
Comment (IshaanAdarsh) [reply Quick-Start/Anatomy_of_extension.md:148]: I have covered this part in the prerequisites, I'll link that part.

File: Quick-Start/Anatomy_of_extension.md (hunk 9 above)
Comment (IshaanAdarsh) [reply Quick-Start/Anatomy_of_extension.md:63]: Yes, I am referring to the `comment` field. 
I thought that the README covered the broader aspects of the extension.
Explaining some commands specific to the control file can be done through `comment`. I should explain that more clearly.

File: Quick-Start/Anatomy_of_extension.md (hunk 10 above)
Comment (IshaanAdarsh) [reply Quick-Start/Anatomy_of_extension.md:65]: I have summarized the above points for easier reviews on the above points (makes it easier I thought not pointing to each separate point just giving a broad problem).
But i am not familiar with PR's inline system, that kind of makes this para useless. i will remove these summary paras in the upcoming iterations

**Pull request merged**